### PR TITLE
Nmc/style sheet changes

### PR DESCRIPTION
**Pull request by TSI-parulsharma**
Did changes related to style sheet.
Added global style sheet file style.qss
Styling toolbar and progress bar using style.qss

### CODE REVIEW

File: resources.qrc [hunk 1]
```diff
@@ -60,4 +60,7 @@
         <file>src/gui/ConflictDelegate.qml</file>
         <file>src/gui/ConflictItemFileInfo.qml</file>
     </qresource>
+	<qresource prefix="/qss">
```
[the remark this answers is not in the document]
Comment (TSI-parulsharma) [reply resources.qrc:63]: I added this qss resource file using Qt Creator add prefix and Add file button.

File: src/gui/settingsdialog.cpp [hunk 2]
```diff
@@ -88,6 +80,10 @@ SettingsDialog::SettingsDialog(ownCloudGui *gui, QWidget *parent)
     _toolBar = new QToolBar;
     _toolBar->setIconSize(QSize(32, 32));
     _toolBar->setToolButtonStyle(Qt::ToolButtonTextUnderIcon);
+    //_toolBar->setProperty(highlightColor, palette().highlight().color().name());
+    //_toolBar->setProperty(highlightTextColor, Theme::defaultColor().name());
+    //_toolBar->setProperty(dark, palette().dark().color().name());
+    //_toolBar->setProperty(background, palette().base().color().name());
```
[the remark this answers is not in the document]
Comment (TSI-parulsharma) [reply src/gui/settingsdialog.cpp]: Fixed

File: src/gui/folderwizard.cpp [hunk 3]
```diff
@@ -643,6 +643,7 @@ FolderWizard::FolderWizard(AccountPtr account, QWidget *parent)
         _folderWizardTargetPage->installEventFilter(this);
     }
     setPage(Page_SelectiveSync, _folderWizardSelectiveSyncPage);
+    this->setObjectName("folderWizard");
```
[the remark this answers is not in the document]
Comment (TSI-parulsharma) [reply src/gui/folderwizard.cpp]: Fixed

File: src/gui/application.cpp [hunk 4]
```diff
@@ -1013,4 +1015,17 @@ bool Application::event(QEvent *event)
...
+{
+    QString highlightColor(palette().highlight().color().name());
+    QString highlightTextColor(palette().highlightedText().color().name());
+    QString dark(palette().dark().color().name());
+    QString background(palette().base().color().name());
+
+    QFile file(":/qss/theme/Style/style.qss");
+    file.open(QFile::ReadOnly);
+    QString styleSheet = QLatin1String(file.readAll());
+    this->setStyleSheet(styleSheet.arg(background, dark, highlightColor, highlightTextColor));
```
[the remark this answers is not in the document]
Comment (TSI-parulsharma) [reply src/gui/application.cpp]: Fixed

File: src/gui/wizard/owncloudwizard.cpp [hunk 5]
```diff
@@ -302,10 +302,17 @@ void OwncloudWizard::slotCurrentPageChanged(int id)
     const auto setNextButtonAsDefault = [this]() {
         auto nextButton = qobject_cast<QPushButton *>(button(QWizard::NextButton));
         if (nextButton) {
+            nextButton->setObjectName("nextButton");
             nextButton->setDefault(true);
         }
     };
 
+    auto backButton = qobject_cast<QPushButton *>(button(QWizard::BackButton));
+    if (backButton) {
```
[the remark this answers is not in the document]
Comment (TSI-parulsharma) [reply src/gui/wizard/owncloudwizard.cpp]: Fixed

File: theme/Style/style.qss [hunk 6]
```diff
@@ -0,0 +1,9 @@
+
+QToolBar { background: %1; margin: 0; padding: 0; border: none; border-bottom: 1px solid %2; spacing: 0; }
+QToolBar QToolButton { background: %1; border: none; border-bottom: 1px solid %2; margin: 0; padding: 5px; }
+QToolBar QToolBarExtension { padding:0; }
+QToolBar QToolButton:checked { background: %3; color: %4; }
+
+
+QProgressBar {border: 1px solid grey; border-radius: 5px; text-align: center;}
```
[the remark this answers is not in the document]
Comment (TSI-parulsharma) [reply theme/Style/style.qss]: Fixed

File: src/gui/application.h [hunk 7]
```diff
@@ -124,6 +124,8 @@ protected slots:
      */
     bool configVersionMigration();
 
+    void setCustomStyleSheet();
```
[the remark this answers is not in the document]
Comment (TSI-parulsharma) [reply src/gui/application.h]: Fixed

File: src/gui/wizard/owncloudwizard.cpp [hunk 8]
```diff
@@ -302,10 +302,17 @@ void OwncloudWizard::slotCurrentPageChanged(int id)
     const auto setNextButtonAsDefault = [this]() {
         auto nextButton = qobject_cast<QPushButton *>(button(QWizard::NextButton));
         if (nextButton) {
```
[the remark this answers is not in the document]
Comment (TSI-parulsharma) [reply src/gui/wizard/owncloudwizard.cpp]: Fixed

File: src/gui/wizard/owncloudwizard.cpp [hunk 9]
```diff
@@ -302,10 +302,17 @@ void OwncloudWizard::slotCurrentPageChanged(int id)
     const auto setNextButtonAsDefault = [this]() {
         auto nextButton = qobject_cast<QPushButton *>(button(QWizard::NextButton));
```
[the remark this answers is not in the document]
Comment (TSI-parulsharma) [reply src/gui/wizard/owncloudwizard.cpp]: Fixed

**Review by mgallien — CHANGES_REQUESTED** [review]
see my inline comments

File: src/gui/application.cpp [hunk 10]
```diff
@@ -1013,4 +1015,19 @@ bool Application::event(QEvent *event)
...
+{
+    const auto wizardPalette = palette();
+    const auto highlightColor = wizardPalette.highlight().color().name();
+    const auto highlightTextColor = wizardPalette.highlightedText().color().name();
+    const auto dark = wizardPalette.dark().color().name();
+    const auto background = wizardPalette.base().color().name();
+
+    QFile file(QStringLiteral(":/qss/theme/Style/style.qss"));
+    if (file.open(QFile::ReadOnly)) {
+        QString styleSheet = QLatin1String(file.readAll());
```
[the remark this answers is not in the document]
Comment (TSI-parulsharma) [reply src/gui/application.cpp]: Yes correct! I fixed it :)

File: src/gui/wizard/owncloudwizard.cpp [hunk 11]
```diff
@@ -300,12 +300,17 @@ void OwncloudWizard::slotCurrentPageChanged(int id)
...
-        auto nextButton = qobject_cast<QPushButton *>(button(QWizard::NextButton));
-        if (nextButton) {
+        if (auto nextButton = qobject_cast<QPushButton *>(button(QWizard::NextButton))) {
+            nextButton->setObjectName("nextButton");
             nextButton->setDefault(true);
         }
     };
 
+    if (auto backButton = qobject_cast<QPushButton *>(button(QWizard::BackButton))) {
+        backButton->setObjectName("backButton");
```
[the remark this answers is not in the document]
Comment (TSI-parulsharma) [reply src/gui/wizard/owncloudwizard.cpp:310]: We want to change the styles of this button as per our customization from style.qss. To access in style.qss, we need indentifier. If we set the object name in NextCloud original code then modification will not be required in these files for customization. We can directly change the styles from style.qss

File: src/gui/wizard/owncloudwizard.cpp [hunk 12]
```diff
@@ -300,12 +300,17 @@ void OwncloudWizard::slotCurrentPageChanged(int id)
...
+        backButton->setDefault(true);
```
[the remark this answers is not in the document]
Comment (TSI-parulsharma) [reply src/gui/wizard/owncloudwizard.cpp:311]: I think we can have two default buttons on QWizardPages. We ran the code and tested properly. We did this only to set the object name to change the default styling of this button. If you want we can remove line 310(backButton->setDefault(true);).

File: src/gui/wizard/owncloudwizard.cpp [hunk 13]
```diff
@@ -300,12 +300,17 @@ void OwncloudWizard::slotCurrentPageChanged(int id)
     qCDebug(lcWizard) << "Current Wizard page changed to " << id;
 
     const auto setNextButtonAsDefault = [this]() {
-        auto nextButton = qobject_cast<QPushButton *>(button(QWizard::NextButton));
-        if (nextButton) {
+        if (auto nextButton = qobject_cast<QPushButton *>(button(QWizard::NextButton))) {
+            nextButton->setObjectName("nextButton");
```
[the remark this answers is not in the document]
Comment (TSI-parulsharma) [reply src/gui/wizard/owncloudwizard.cpp:304]: We want to change the styles of this button as per our customization from style.qss. To access in style.qss, we need indentifier.

File: src/gui/folderwizard.cpp [hunk 14]
```diff
@@ -643,6 +643,7 @@ FolderWizard::FolderWizard(AccountPtr account, QWidget *parent)
         _folderWizardTargetPage->installEventFilter(this);
     }
     setPage(Page_SelectiveSync, _folderWizardSelectiveSyncPage);
+    setObjectName("folderWizard");
```
[the remark this answers is not in the document]
Comment (TSI-parulsharma) [reply src/gui/folderwizard.cpp:646]: We want to change the styles of this button as per our customization from style.qss. To access in style.qss, we need indentifier.

**Review by mgallien — APPROVED** [review]

**Comment by mgallien** [comment]
@TSI-parulsharma sorry for the delay, this slipped out of my mind
we have a DCO check that is failing
see https://github.com/nextcloud/desktop/pull/5776/checks?check_run_id=14788692861
I can help if needed

**Comment by tsdicloud** [comment]
Does anybody understand the purpose of this PR?
I think it missed the actual topic of more flexible theming...

Any idea how to proceed?

**Comment by tsdicloud** [comment]
I don't understand why individual changes from MagentaCLOUD should be upstreamed.
The stylesheet mechanism is maybe nice (although I am not completely convinced anymore as it is only a complicated version of and added QStyle class...

MagentaCLOUD should not upstream things that may need to change on demand, as it would then mean to always ask for an upstream change. Let's discuss this F2F as planned.

**Pull request closed without merging**